### PR TITLE
Chat 185 - Expose Timetoken on Emit

### PR DESCRIPTION
**Pull request by ianjennings**
This makes a ```new Event``` for every publish so that we can track the network status of that communication. 

For example, here is how to grab the timetoken response on a successful publish:

```js
        let chat = new ChatEngine.Chat('chat-tester' + new Date().getTime());

        let event = chat.emit('test');

        event.on('$.emitted', (a) => {
            assert(a.timetoken, 'Timetoken exposed on emit');
            console.log(a.timetoken);
            done();
        });
```

### CODE REVIEW

File: src/components/chat.js [hunk 1]
```diff
@@ -310,12 +312,13 @@ class Chat extends Emitter {
             // publish the event and data over the configured channel
 
             // ensure the event exists within the global space
```
Comment (crgbaumgart) [inline src/components/chat.js]: Is the above comment stale now? Seems to be.

File: src/components/event.js [hunk 2]
```diff
@@ -60,23 +64,28 @@ class Event {
...
 
             if (status.statusCode === 200) {
 
+                if (response) {
+                    m.timetoken = response.timetoken;
+                }
+
                 /**
                  * Message successfully published
                  * @event Chat#$"."publish"."success
```
Comment (crgbaumgart) [inline src/components/event.js]: Another comment nitpick, wouldn't the ```@event Chat#$"."publish"."success``` look more like ```@event Chat#$."emitted```?

Comment (ianjennings) [reply src/components/event.js]: yep

File: src/components/chat.js (hunk 1 above)
Comment (ianjennings) [reply src/components/chat.js]: yep

**Comment by ianjennings** [comment]
Tests are throwing a lot of these:

(node:43922) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 88): TypeError: Converting circular structure to JSON

**Comment by ianjennings** [comment]
This will have collision with #302 Event class

**Comment by ianjennings** [comment]
@crgbaumgart added more docs / comments

**Pull request merged**